Description: handle the 'trusty' qemu machine type
 This treats the 'trusty' machine type the same way as pc-i440.
Author: Serge Hallyn <serge.hallyn@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/qemu/+bug/1294823
Forwarded: no

Index: libvirt-1.2.2/src/qemu/qemu_command.c
===================================================================
--- libvirt-1.2.2.orig/src/qemu/qemu_command.c	2014-02-21 11:24:40.000000000 +0000
+++ libvirt-1.2.2/src/qemu/qemu_command.c	2014-03-25 17:11:28.763612517 +0000
@@ -2521,7 +2521,8 @@
             STRPREFIX(def->os.machine, "pc-0.") ||
             STRPREFIX(def->os.machine, "pc-1.") ||
             STRPREFIX(def->os.machine, "pc-i440") ||
-            STRPREFIX(def->os.machine, "rhel"));
+            STRPREFIX(def->os.machine, "rhel") ||
+            STREQ(def->os.machine, "trusty"));
 }
 
 
@@ -2721,6 +2722,7 @@
         STRPREFIX(def->os.machine, "pc-1.") ||
         STRPREFIX(def->os.machine, "pc-i440") ||
         STREQ(def->os.machine, "pc") ||
+        STREQ(def->os.machine, "trusty") ||
         STRPREFIX(def->os.machine, "rhel")) &&
         qemuValidateDevicePCISlotsPIIX3(def, qemuCaps, addrs) < 0) {
         goto error;
Index: libvirt-1.2.2/src/qemu/qemu_domain.c
===================================================================
--- libvirt-1.2.2.orig/src/qemu/qemu_domain.c	2014-02-21 11:24:40.000000000 +0000
+++ libvirt-1.2.2/src/qemu/qemu_domain.c	2014-03-25 17:10:46.503664072 +0000
@@ -720,7 +720,8 @@
             !STRPREFIX(def->os.machine, "pc-1.") &&
             !STRPREFIX(def->os.machine, "pc-i440") &&
             !STREQ(def->os.machine, "pc") &&
-            !STRPREFIX(def->os.machine, "rhel"))
+            !STRPREFIX(def->os.machine, "rhel") &&
+            !STREQ(def->os.machine, "trusty"))
             break;
         addPCIRoot = true;
         break;
